### ISSUE TITLE
sprintf_linter causes test warning on r-devel

### ISSUE THREAD

**Issue by AshesITR** [jimhester/lintr#624]
https://travis-ci.org/github/jimhester/lintr/jobs/746547257#L1728

```
── Warning (test-sprintf_linter.R:7:3): returns the correct linting ────────────

one argument not used by format 'hello'

── Warning (test-sprintf_linter.R:43:3): returns the correct linting ───────────

one argument not used by format 'hello %1$s %s'

── Warning (test-sprintf_linter.R:47:3): returns the correct linting ───────────

one argument not used by format 'hello %1$s %1$s'
```

Should catch the warning in the linter using `withCallingHandlers` and turn it into a lint.
